### ISSUE TITLE
False positive in Nullability analysis within OR conditional 

### ISSUE THREAD

**Issue by buyaa-n** [dotnet/roslyn#70962]
This is failing roslyn => runtime [dependency flow](https://github.com/dotnet/runtime/pull/94611). Have no idea why this failure is happening now, the [code that failing](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs#L2076) did not changed for many years and the nullability analysis enabled for the System.Private.XML project several years ago. Might something changed in roslyn.

```log
src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs(2076,48): error CS8601: (NETCORE_ENGINEERING_TELEMETRY=Build) Possible null reference assignment.
src/libraries/System.Private.Xml/src/System/Xml/Schema/SchemaCollectionCompiler.cs(2080,75): error CS8604: (NETCORE_ENGINEERING_TELEMETRY=Build) Possible null reference argument for parameter 's' in 'object XmlSchemaDatatype.ParseValue(string s, XmlNameTable? nameTable, IXmlNamespaceResolver? nsmgr, bool createAtomicValue)'
```

**Steps to Reproduce**:

```cs
internal string s;
internal void NullabilityTest(string? a, string? b)
{
    if (a != null || b != null)
    {
        if (a != null)
        {
            s = a;
        }
        else
        {
            s = b; // warns 
        }
    }

    int i = int.Parse(s); // warns 
}
```

**Diagnostic Id**:

CS8601, CS8604

**Expected Behavior**:
Now warning

**Actual Behavior**:
Warns

**Comment by CyrusNajmabadi** [comment]
This is expected relationships between variables are not tracked by the compiler

**Comment by buyaa-n** [comment]
If it's not fixable feel free to close
